core/src/worker/workflow/history_update.rs

```diff
@@ -408,7 +408,10 @@ impl Stream for StreamingHistoryPaginator {
             // SAFETY: This is safe because the inner paginator cannot be dropped before the future,
             //   and the future won't be moved from out of this struct.
             this.open_history_request.set(Some(unsafe {
-                transmute(HistoryPaginator::get_next_page(this.inner).boxed())
+                transmute::<
```

Comment:
```rust
djc-2021 act-opts-tq-option temporal-sdk $ cargo lint
    Checking temporal-sdk-core v0.1.0 (/Users/djc/src/temporal-sdk/core)
error: transmute used without annotations
   --> core/src/worker/workflow/history_update.rs:411:17
    |
411 |                 transmute::<_, BoxFuture<'static, Result<(), tonic::Status>>>(
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider adding missing annotations: `transmute::<std::pin::Pin<std::boxed::Box<dyn futures_util::Future<Output = std::result::Result<(), tonic::Status>> + std::marker::Send>>, std::pin::Pin<std::boxed::Box<dyn futures_util::Future<Output = std::result::Result<(), tonic::Status>> + std::marker::Send>>>`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_transmute_annotations
    = note: `-D clippy::missing-transmute-annotations` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::missing_transmute_annotations)]`

error: could not compile `temporal-sdk-core` (lib) due to 1 previous error
```

Which is an interesting suggestion because this was previously transmuting a `BoxFuture<'_, Result<bool, tonic::Status>>` to `BoxFuture<'static, Result<(), tonic::Status>>` which I'm guessing was a bit more than intentional.

Reply:
Huh, weird.